### PR TITLE
Improve AOT unit tests

### PR DESCRIPTION
**Pull request by tbosch**

### CODE REVIEW

File: packages/compiler/src/jit/compiler.ts [hunk 1]
```diff
@@ -74,10 +75,25 @@ export class JitCompiler {
...
     this.clearCache();
-    flattenSummaries(summaries).forEach((summary) => {
-      this._summaryResolver.addSummary(
-          {symbol: summary.type.reference, metadata: null, type: summary});
-    });
+    this._addAotSummaries(summaries);
+  }
+
+  private _addAotSummaries(fn: () => any[]) {
+    if (this._addedAotSummaries.has(fn)) {
```
Comment (vicb) [inline packages/compiler/src/jit/compiler.ts]: Do we want to marked only non-functions as visited ? 
Could help with stateful functions ? (not sure about the exact use case for fns)

Comment (vicb) [reply packages/compiler/src/jit/compiler.ts]: Oh this was existing code... so probably not worth changing the behavior

File: packages/core/testing/src/test_bed.ts [hunk 2]
```diff
@@ -478,6 +500,16 @@ export class TestBed implements Injector {
     overrideProvider({token, flags, deps, value, deprecatedBehavior: deprecated});
   }
 
+  overrideTemplateUsingTestingModule(component: Type<any>, template: string) {
+    this._assertNotInstantiated('overrideTemplateUsingTestingModule', 'override template');
+
+    @Component({selector: 'empty', template: template})
```
Comment (vicb) [inline packages/core/testing/src/test_bed.ts]: nit `template: template` could be simplified

File: packages/platform-browser/test/testing_public_spec.ts [hunk 3]
```diff
@@ -712,6 +712,60 @@ export function main() {
...
+          }
+
+          TestBed.get(Injector);
+
+          expect(() => TestBed.overrideTemplateUsingTestingModule(MyComponent, 'b'))
+              .toThrowError(
+                  /Cannot override template when the test module has already been instantiated/);
+        });
+
+        it('should reset overrides when the testing modules is resetted', () => {
```
Comment (vicb) [inline packages/platform-browser/test/testing_public_spec.ts]: testing module-s-

**Review by vicb — CHANGES_REQUESTED** [review]
see inline comments

**Comment by tbosch** [comment]
FYI: @vicb is taking over this PR and landing it once Angular 5 is done.

**Comment by tbosch** [comment]
@vicb will take over this PR.

File: packages/core/src/view/services.ts [hunk 4]
```diff
@@ -139,10 +143,15 @@ function debugCreateEmbeddedView(
 
 function debugCreateComponentView(
     parentView: ViewData, nodeDef: NodeDef, viewDef: ViewDefinition, hostElement: any): ViewData {
-  const defWithOverride = applyProviderOverridesToView(viewDef);
+  const overrideComponentView =
```
[the remark this answers is not in the document]
Comment (vicb) [reply packages/core/src/view/services.ts]: we already have `OverrideProvider`, keeping for consistency

**Comment by IgorMinar** [comment]
closing in favor of #20224

**Pull request closed without merging**